src/kernel-install/60-ukify.install.in

```diff
@@ -0,0 +1,189 @@
+    if VERBOSE:
+        print(*args, **kwargs)
+
+def path_is_readable(p: Path, dir=False) -> None:
+    """Verify access to a file or directory."""
+    try:
+        p.open().close()
+    except IsADirectoryError:
+        if dir:
+            return
+        raise
+
+def parse_args(args=None):
+    p = argparse.ArgumentParser(
+        description='kernel-install plugin to build a Unified Kernel Image',
+        allow_abbrev=False,
+        usage='''\
+usage: 60-ukify.install COMMAND KERNEL_VERSION ENTRY_DIR KERNEL_IMAGE INITRD…
+       60-ukify.install -h | --help
+''')
+
+    # Suppress printing of usage synopsis on errors
+    p.error = lambda message: p.exit(2, f'{p.prog}: error: {message}\n')
+
+    p.add_argument('command',
+                   metavar='COMMAND',
+                   help="The action to perform. Only 'add' is supported.")
+    p.add_argument('kernel_version',
+                   metavar='KERNEL_VERSION',
+                   help='Kernel version string')
+    p.add_argument('entry_dir',
+                   metavar='ENTRY_DIR',
+                   type=Path,
+                   nargs='?',
+                   help='Type#1 entry directory (ignored)')
+    p.add_argument('kernel_image',
+                   metavar='KERNEL_IMAGE',
+                   nargs='?',
+                   help='Kernel binary')
+    p.add_argument('initrd',
+                   metavar='INITRD…',
+                   nargs='*',
+                   help='Initrd files')
+    p.add_argument('--version',
+                   action='version',
+                   version=f'systemd {__version__}')
+
+    opts = p.parse_args(args)
+
+    if opts.command == 'add':
+        opts.staging_area = Path(os.environ['KERNEL_INSTALL_STAGING_AREA'])
+        path_is_readable(opts.staging_area, dir=True)
+
+    return opts
+
+def we_are_wanted() -> bool:
+    KERNEL_INSTALL_LAYOUT = os.getenv('KERNEL_INSTALL_LAYOUT')
+
+    if KERNEL_INSTALL_LAYOUT != 'uki':
+        log(f'{KERNEL_INSTALL_LAYOUT=}, quitting.')
+        return False
+
+    KERNEL_INSTALL_UKI_GENERATOR = os.getenv('KERNEL_INSTALL_UKI_GENERATOR')
+
+    if KERNEL_INSTALL_UKI_GENERATOR != 'ukify':
+        log(f'{KERNEL_INSTALL_UKI_GENERATOR=}, quitting.')
+        return False
+
+    log('KERNEL_INSTALL_LAYOUT and KERNEL_INSTALL_UKI_GENERATOR are good')
+    return True
+
+def kernel_cmdline() -> str:
+    entry_token = os.environ['KERNEL_INSTALL_ENTRY_TOKEN']
+    machine_id = os.environ['KERNEL_INSTALL_MACHINE_ID']
+
+    if root := os.getenv('KERNEL_INSTALL_CONF_ROOT'):
+        try:
+            options = Path(root).joinpath('cmdline').read_text().split()
+        except FileNotFoundError:
+            return None
+    else:
+        for cmdline, fallback in [
+                ('/etc/kernel/cmdline',     False),
+                ('/usr/lib/kernel/cmdline', False),
+                ('/proc/cmdline',           True),
+        ]:
+            try:
+                options = Path(cmdline).read_text().split()
```

Comment:
doesn't this lack a "break" somewhere? otherwise it will read all three files one after the other, and ignore the first two after reading them..

Reply:
It does. Thanks for catching this.

But after looking at this again, I reworked this to have a separate function for reading of the file and changed the loop to only cover the first two cases, and have the third path in a separate chunk. It seems much more legible this way.